src/main/java/com/google/api/codegen/config/PageStreamingConfig.java

```diff
@@ -176,18 +178,40 @@ static PageStreamingConfig createPageStreaming(
               "Response field missing for page streaming: method = %s, message type = %s, field = %s",
               method.id(),
               method.id(),
-              responseTokenFieldName));
+              PARAMETER_NEXT_PAGE_TOKEN));
     }
 
     Schema responseField = method.response().dereference();
-    Schema resourcesField = null;
+    DiscoveryField resourcesField = null;
+    List<FieldModel> resourcesFieldPath = new LinkedList<>();
     for (Schema property : responseField.properties().values()) {
       // Assume the List response has exactly one Array property.
       if (property.type().equals(Type.ARRAY)) {
-        resourcesField = property;
+        resourcesField = new DiscoveryField(property, discoGapicNamer);
+        resourcesFieldPath.add(resourcesField);
         break;
+      } else if (property.additionalProperties() != null
+          && !Strings.isNullOrEmpty(property.additionalProperties().reference())) {
+        Schema additionalProperties = property.additionalProperties().dereference();
+        if (additionalProperties.type().equals(Type.ARRAY)) {
+          resourcesField = new DiscoveryField(additionalProperties, discoGapicNamer);
+          resourcesFieldPath.add(resourcesField);
+          break;
+        }
+        for (Schema subProperty : additionalProperties.properties().values()) {
+          if (subProperty.type().equals(Type.ARRAY)) {
+            resourcesField = new DiscoveryField(subProperty, discoGapicNamer);
+            resourcesFieldPath.add(new DiscoveryField(property, discoGapicNamer));
+            resourcesFieldPath.add(resourcesField);
+            break;
+          }
+        }
+        if (resourcesField != null) {
+          break;
+        }
       }
     }
+
```

Comment:
This loop got big, could you pull it out to a helper method?

Reply:
Done.